tools/ctl/ide/rebuilder.go

```diff
@@ -194,27 +197,47 @@ func (rb *Rebuilder) Restart(ctx context.Context) {
 	}
 }
 
+type RunLocalOpts struct {
+	Strategy *schema.StrategyOneOf
+}
+
 // RunLocal runs the rebuilder for the given example.
 // Each element of extraParams is a url parameter like "param=value".
-func (rb *Rebuilder) RunLocal(ctx context.Context, r firestore.Rebuild, extraParams ...string) {
+func (rb *Rebuilder) RunLocal(ctx context.Context, r firestore.Rebuild, opts RunLocalOpts) {
 	_, err := rb.runningInstance(ctx)
 	if err != nil {
-		log.Println(err)
+		log.Println(err.Error())
 		return
 	}
 	log.Printf("Calling the rebuilder for %s\n", r.ID())
 	id := time.Now().UTC().Format(time.RFC3339)
-	var extra string
-	if len(extraParams) > 0 {
-		extra = "&" + strings.Join(extraParams, "&")
+	vals := url.Values{}
+	vals.Add("ecosystem", r.Ecosystem)
+	vals.Add("pkg", r.Package)
+	vals.Add("versions", r.Version)
+	vals.Add("id", id)
+	if opts.Strategy != nil {
+		byts, err := json.Marshal(opts.Strategy)
+		if err != nil {
+			log.Println(err)
+			return
+		}
+		vals.Add("strategy", string(byts))
 	}
-	cmd := exec.CommandContext(ctx, "curl", "--silent", "-d", fmt.Sprintf("ecosystem=%s&pkg=%s&versions=%s&id=%s%s", r.Ecosystem, r.Package, r.Version, id, extra), "-X", "POST", "127.0.0.1:8080/smoketest")
-	rllog := logWriter(log.New(log.Default().Writer(), logPrefix("runlocal"), 0))
-	cmd.Stdout = rllog
-	cmd.Stderr = rllog
-	log.Println(cmd.String())
-	if err := cmd.Start(); err != nil {
-		log.Println(err)
+	client := http.DefaultClient
+	resp, err := client.PostForm("http://127.0.0.1:8080/smoketest", vals)
+	if err != nil {
+		log.Println(err.Error())
+		return
+	}
+	var smkResp schema.SmoketestResponse
+	if err := json.NewDecoder(resp.Body).Decode(&smkResp); err != nil {
+		log.Println(errors.Wrap(err, "failed to decode smoketest response").Error())
+	}
+	// TODO: Is there a more usefull way to print this response?
+	log.Println(smkResp)
+	if len(smkResp.Verdicts) == 0 && smkResp.Verdicts[0].Message == "" {
```

Comment:
`len(smkResp.Verdicts) != 0` ? this predicate seems odd

Reply:
Fixed.